prowler/providers/azure/azure_provider.py

```diff
@@ -628,7 +629,7 @@ async def get_azure_identity():
             logger.critical(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}] -- {error}"
             )
-            sys.exit(1)
+            raise error
```

Comment:
Add a TODO here to replace it with our custom exceptions once they are ready.

Reply:
Done